src/main/cpp/src/get_json_object.cu

```diff
@@ -660,8 +377,7 @@ __device__ inline thrust::tuple<bool, int> path_match_subscript_index_subscript_
 __device__ bool evaluate_path(json_parser& p,
                               json_generator& root_g,
                               write_style root_style,
-                              path_instruction const* root_path_ptr,
-                              int root_path_size)
+                              cudf::device_span<path_instruction const> root_path)
```

Comment:
Note sure if using device_span will cause perf regression, because the `size` in `device_span` is 64 bits integer which is using more memory compared to 32 bits size.

Reply:
no perf regression seen from microbenchmark